File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/registry/impl/EurekaRegisterFactory.java
package org.javamaster.invocationlab.admin.service.registry.impl;

import org.javamaster.invocationlab.admin.service.registry.Register;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

/**
 * @author yudong
 */
@Component
public class EurekaRegisterFactory extends AbstractRegisterFactory {
    @Autowired
    private RestTemplate restTemplate;

    @Override
    public Register create(String cluster) {
        return new EurekaRegister(cluster, restTemplate);
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/registry/impl/EurekaRegisterFactory.kt
package org.javamaster.invocationlab.admin.service.registry.impl

import org.javamaster.invocationlab.admin.service.registry.Register
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

/**
 * @author yudong
 */
@Component
class EurekaRegisterFactory : AbstractRegisterFactory() {
    @Autowired
    private lateinit var restTemplate: RestTemplate

    override fun create(cluster: String): Register {
        return EurekaRegister(cluster, restTemplate)
    }
}
